### PR TITLE
Update breaking changes document for 17.0 struct initializer fixes

### PR DESCRIPTION
**Pull request by cston**
Include specific breaking change entries corresponding to https://github.com/dotnet/roslyn/pull/59055.

Also, added anchor tags to each breaking change entry.

### CODE REVIEW

**Review by 333fred — APPROVED** [review]

File: docs/compilers/CSharp/Compiler Breaking Changes - DotNet 6.md [hunk 1]
```diff
@@ -1,14 +1,15 @@
 ﻿## This document lists known breaking changes in Roslyn in C# 10.0 which will be introduced with .NET 6.
 
-1. Beginning with C# 10.0, null suppression operator is no longer allowed in patterns.
+1. <a name="1"></a>Beginning with C# 10.0, null suppression operator is no longer allowed in patterns.
```
Comment (Youssef1313) [inline docs/compilers/CSharp/Compiler Breaking Changes - DotNet 6.md:3]: Why do these anchor tags exist?

Comment (cston) [reply docs/compilers/CSharp/Compiler Breaking Changes - DotNet 6.md:3]: To allow links to specific items.

File: docs/compilers/CSharp/Compiler Breaking Changes - DotNet 6.md [hunk 2]
```diff
@@ -109,13 +110,54 @@ These are _function_type_conversions_.
     }
     ```
 
-4. In Visual Studio 17.1, `struct` type declarations with field initializers must include an explicitly declared constructor. Additionally, all fields must be definitely assigned in `struct` instance constructors that do not have a `: this()` initializer so any previously unassigned fields must be assigned from the added constructor or from field initializers.
+4. <a name="4"></a>In Visual Studio 17.0 servicing, an error is reported in a `record struct` with a primary constructor if an explicit constructor has a `this()` initializer that invokes the implicit parameterless constructor.
```
Comment (jcouv) [inline docs/compilers/CSharp/Compiler Breaking Changes - DotNet 6.md:113]: nit: consider add links to an issue or PR if it might be useful reference.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 5)

**Pull request merged**